doc/api/stream.md

```diff
@@ -154,89 +325,134 @@ Examples of readable streams include:
 * [child process stdout and stderr][]
 * [`process.stdin`][]
 
-#### Event: 'close'
+All [Readable][] streams implement the interface defined by the
+`stream.Readable` class.
+
+#### Two Modes
+
+Readable streams may operate in one of two modes: flowing and paused.
```

Comment:
This is kind of a tough topic because it unearths a bit of the underlying streams (and in particular, `pipe()`'s) machinery. It treats `paused` as a distinct state that a stream can be shifted into by the user as well, when in reality there's just the tri-state property `._readableState.flowing`.

It might be best to illustrate this with a directed graph?

![output](https://cloud.githubusercontent.com/assets/37303/15495521/5b664ef6-2145-11e6-993a-1325c652306e.png)

The notable thing is that streams, by default, won't generate data until you've provided a mechanism to take ownership of that data, or have indicated that you don't care about it using `.resume()`. If you take away the mechanism for ownership, streams will attempt to stop generating data.